nixos/modules/config/xdg/portal.nix

```diff
@@ -54,13 +54,18 @@ with lib;
     in mkIf cfg.enable {
 
       assertions = [
-        { assertion = (cfg.gtkUsePortal -> cfg.extraPortals != []);
-          message = "Setting xdg.portal.gtkUsePortal to true requires a portal implementation in xdg.portal.extraPortals such as xdg-desktop-portal-gtk or xdg-desktop-portal-kde.";
+        { assertion = cfg.extraPortals != [];
+          message = "Setting xdg.portal.enable to true requires a portal implementation in xdg.portal.extraPortals such as xdg-desktop-portal-gtk or xdg-desktop-portal-kde.";
         }
       ];
 
       services.dbus.packages  = packages;
       systemd.packages = packages;
+      # fixes screen sharing on non-chromium apps by linking
+      # share/applications/org.freedesktop.impl.portal.desktop.*.desktop files
```

Comment:
Only KDE portal uses this naming scheme. All other portal implementations either use unqualified one (`xdg-desktop-portal-gtk`, `xdg-desktop-portal-gnome`) or do not ship desktop file at all (`pantheon.xdg-desktop-portal-pantheon`, `xdg-desktop-portal-wlr`).